GUIDELINES.md

```diff
@@ -0,0 +1,892 @@
+
+If you're wondering how the `BackendFakeUser` looks like, here it is:
+
+```kotlin
+data class BackendFakeUser(
+    val email: String,
+    val password: String,
+    val authToken: AuthToken = UUID.randomUUID().toString()
+) {
+    companion object {
+        val USER_A = BackendFakeUser("user.a@test.com", "supersecure_a")
+        val USER_B = BackendFakeUser("user.b@test.com", "supersecure_b")
+    }
+}
+```
+
+It also contains predefined data `USER_A` and `USER_B` so it can be reused by multiple tests without the need to define constants each time. Maybe you noticed that this class is outside the steps class; that is intentional: its use reaches beyond the steps class as it is handy for various test scenarios, no matter you decide to use the fake backend or not.
+
+#### Create the steps class for the unit test
+
+So let's create a steps class responsible for just unit-testing `AuthManager`:
+
+```kotlin
+class AuthManagerSteps(testContext: TestContext) : BaseSteps(testContext, appModuleTestingConfiguration) {
+
+    val backend by steps<BackendFakeSteps>()
+
+    private val instance by dependency<AuthManager>()
+    private val sessionStore by dependency<SessionStore>()
+
+    override fun configure() = super.configure()
+        .requireReal<AuthManager>()
+}
+```
+
+As we anticipate that the fake backend and the `SessionStore` will be needed afterwards (they are the two direct dependencies of `AuthManager` which are consumed by its constructor).
+
+There is only one `requireReal` call (`requireReal<AuthManager>()`), which clearly indicates that this steps class is indeed about unit-testing just the `AuthManager`.
+
+But why is `backend` just called `backend` and not `backendGateway`? And why is it public? Shouldn't a steps class be a full abstraction of what's going on under the hood or encapsulate everything needed for the test class using it? Generally yes. Usually we should avoid offering internals ([Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter)). Access to internals should be done through public accessors in order to avoid tying code to details which are prone to change. But anyway, the key here is "prone to change": if you group and define systems around abstract business concepts these are very unlikely to change (contrary to technical implementation). This makes the fake backend a better candidate for being used and shared among various tests. So in this case it is reasonable to call the member after the abstract concept of a "backend" _and_ offering it via the steps classes' API for direct use in test classes.
```

Comment:
i agree that this touches quite a lot. it is very important for me to get the whole abstraction topic across, as it's the whole point of sweetest, right? this was the original intention behind this sentence. could you offer me a better way of achieving this?

Reply:
> it is very important for me to get the whole abstraction topic across, as it's the whole point of sweetest, right?

Fair enough. I would kept these ideas still more marked as tip. As especially for the introduction it might be a bit overwhelming. My suggestion was the one sentence in my comment, but up to you.

Reply:
hm, i think i found a way to accomodate your feedback here: i moved the monster sentence to the reference section (namely it fits perfectly in the "Can there be too much abstraction?" chapter).